modules/services/pantalaimon.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.pantalaimon;
+
+  iniFmt = pkgs.formats.ini { };
+in {
+  meta.maintainers = [ maintainers.jojosch ];
+
+  options = {
+    services.pantalaimon = {
+      enable = mkEnableOption
+        "Pantalaimon, an E2EE aware proxy daemon for matrix clients";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.pantalaimon;
+        defaultText = literalExample "pkgs.pantalaimon";
+        description = "The package to use for pantalaimon.";
+      };
+
+      settings = mkOption {
+        type = iniFmt.type;
+        default = { };
+        defaultText = literalExample "{ }";
+        example = literalExample ''
+          {
+            Default = {
+              LogLevel = "Debug";
+              SSL = true;
+            };
+            local-matrix = {
+              Homeserver = "https://matrix.org";
+              ListenAddress = "127.0.0.1";
+              ListenPort = 8008;
+            };
+          }
+        '';
+        description = ''
+          Configuration written to
+          <filename>$XDG_CONFIG_HOME/pantalaimon/pantalaimon.conf</filename>.
+          </para><para>
+          See <link xlink:href="https://github.com/matrix-org/pantalaimon/blob/master/docs/man/pantalaimon.5.md" /> for the full list
+          of options.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.pantalaimon ];
+
+    xdg.configFile."pantalaimon/pantalaimon.conf" = mkIf (cfg.settings != { }) {
+      source = iniFmt.generate "pantalaimon.conf" cfg.settings;
+    };
```

Comment:
I actually think it's better to use the `-c` option for this and pass the config straight to `pantalaimon`. The problem with this method is that HM doesn't automatically detect that the service config has changed, so it doesn't recommend to restart pantalaimon.

```suggestion
```

Reply:
Agreed, if you can manage without placing files in `$HOME` then that is always nice.

Also, fortunately not necessary here, the standard solution to make a systemd unit depend on a different file is to use `Unit.X-Restart-Triggers` like so:

https://github.com/nix-community/home-manager/blob/2a4ab0d891a59fd3a0fc09e9805aad5a8f82dfac/modules/services/polybar.nix#L201-L202

This will be sufficient to cause a unit restart when necessary.

Reply:
Yeah, that makes sense. 

How should I adjust the test files expecting the config at `$XDG_CONFIG_HOME/pantalaimon/pantalaimon.conf`? Just be satisfied if the `$serviceFile` exists and do something like `assertFileRegex $serviceFile 'ExecStart=.*/bin/pantalaimon -c /nix/store/.*-pantalaimon.conf'`?

Reply:
Just saw your comment @rycee, will do the `X-Restart-Triggers` solution.

Reply:
I think that @rycee was saying to use `X-Restart-Triggers` only if absolutely necessary, and to use `-c` when possible?